src/syscheckd/src/create_db.c

```diff
@@ -234,22 +230,19 @@ static void dbsync_attributes_json(const cJSON *dbsync_event, const directory_t
 
     if (configuration->options & CHECK_OWNER) {
         if (aux = cJSON_GetObjectItem(dbsync_event, "uid"), aux != NULL) {
-            char * buffer;
-            os_malloc(OS_SIZE_64, buffer);
-            snprintf(buffer, OS_SIZE_64, "%d", aux->valueint);
-            cJSON_AddStringToObject(attributes, "uid", buffer);
-            os_free(buffer);
+            char *uid = cJSON_GetStringValue(aux);
+            if (uid != NULL && *uid != '\0') {
+                cJSON_AddStringToObject(attributes, "uid", cJSON_GetStringValue(aux));
```

Comment:
Why are we repeating `cJSON_GetStringValue()` instead of reusing `uid`?

```suggestion
                cJSON_AddStringToObject(attributes, "uid", uid);
```